lib/system/cellseqs_v2.nim

```diff
@@ -16,20 +16,21 @@ type
     len, cap: int
     d: CellArray[T]
 
-proc add[T](s: var CellSeq[T], c: T; t: PNimTypeV2) {.inline.} =
+proc resize[T](s: var CellSeq[T]) =
+  s.cap = s.cap * 3 div 2
+  var newSize = s.cap * sizeof(CellTuple[T])
```

Comment:
`newSize` is only used once and its tiny, I would just inline it directly IMHO.

Reply:
Its used on line 25 and 29 🥲 

Inlining the variable changes line 25 to 
```
s.d = cast[CellArray[T]](allocShared(s.cap * sizeof(CellTuple[T])))
``` 
and line 29 to 
```
s.d = cast[CellArray[T]](realloc(s.d, s.cap * sizeof(CellTuple[T])))
```
I'm a fan of the readability improvement. 

Is this a dealbreaker?

Reply:
If we're bikeshedding or code golfing it, a template is more concise than manually in-lining. still self documenting and less redundant.

Reply:
```nim
template calcNewSize(s: CellSeq): int =
  s.cap * sizeof(CellTuple[s.type])

proc resize[T](s: var CellSeq[T]) =
  s.cap = s.cap * 3 div 2

  when compileOption("threads"):
    let oldData = s.d
    s.d = cast[CellArray[T]](allocShared(calcNewSize(s)))
    copyMem(s.d, oldData, s.len * sizeof(CellTuple[T]))
    deallocShared(oldData)
  else:
    s.d = cast[CellArray[T]](realloc(s.d, calcNewSize(s)))
```
I think using a variable is more readable than a template, but the difference in readability likely comes down to developer preference. I'd be fine with this change.

I'm not sure if either of you want me to refactor this code - but now there's 3 differing opinions on how to represent the expression `s.cap * sizeof(CellTuple[T])`.

Reply:
Like i said *If we're bike shedding* I could not careless. :smile:

Reply:
@beef331 @juancarlospaco can I get an approval or rereview then? This is the most confusing PR I have ever created.

Reply:
Hey it says "collaborator" not "contributor" next to my name. :stuck_out_tongue: 
The code seems fine to me, but it does not matter what I say.

Reply:
> Its used on line 25 and 29 smiling_face_with_tear

If you have `when` branches then only 1 will actually exist whatsoever.

`resize` should be a `template` IMHO.

Reply:
> If you have `when` branches then only 1 will actually exist whatsoever.
> 
> `resize` should be a `template` IMHO.

Templates are useful when you want to generate code at compile time by substituting patterns. They are helpful for avoiding code duplication, improving performance by removing function call overhead, and customizing behavior through parameters.

In the case of the `resize` function, the expression `s.cap * sizeof(CellTuple[T])` is only used once, and it doesn't contain any complex logic that could benefit from being extracted into a separate template. Therefore creating a template for this single expression doesn't make sense since there isn't a clear benefit from creating it.

Reply:
Just variable seems fine here?